### PR TITLE
Re-write the way to read the config file for virt-who testing

### PR DESCRIPTION
**Pull request by hkx303**
Hello, would like to merge this PR to the branch: [new_struc_for_virtwho_plugin](https://github.com/SatelliteQE/robottelo/tree/new_struc_for_virtwho_plugin) 

**Description**
#7871 
1. Create a new config file for virt-who hypervisors (**virtwho.properties**)
```
[esx]
hypervisor_type=esx
hypervisor_server=
hypervisor_username=
hypervisor_password=
guest=
guest_port=
guest_username=
guest_password=

[xen]
hypervisor_type=xen
hypervisor_server=
hypervisor_username=
hypervisor_password=
guest=
guest_port=
guest_username=
guest_password=

[hyperv]
hypervisor_type=
hypervisor_server=
hypervisor_username=
hypervisor_password=
guest=
guest_port=
guest_username=
guest_password=

[rhevm]
hypervisor_type=rhevm
hypervisor_server=
hypervisor_username=
hypervisor_password=
guest=
guest_port=
guest_username=
guest_password=

[libvirt]
hypervisor_type=libvirt
hypervisor_server=
hypervisor_username=
hypervisor_password=
guest=
guest_port=
guest_username=
guest_password=

[kubevirt]
hypervisor_type=kubevirt
hypervisor_server=
hypervisor_username=
hypervisor_password=
hypervisor_config_file=/root/kube.conf
guest=
guest_port=
guest_username=
guest_password=

[sku]
vdc_physical=RH00001
vdc_virtual=RH00049
```
2. Create the config/virtwho.py to read the settings for the config file
3. Update all test cases for virt-who plugin based on the new struc.

**Test Result**

```
(venv) [hkx303@kuhuang foreman]$ pytest -v virtwho/ --disable-pytest-warnings --junit-xml=/home/hkx303/Documents/CI/robottelo/robottelo/tests/foreman/test_report.xml -o junit_suite_name=report_xml
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- /home/hkx303/Documents/CI/robottelo/venv/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting 14 items                                                                          2020-08-19 04:50:36 - conftest - DEBUG - Collected 78 test cases
collected 78 items                                                                           

virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_deploy_configure_by_id PASSED [  1%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_deploy_configure_by_script PASSED [  2%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_debug_option PASSED    [  3%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_interval_option PASSED [  5%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_hypervisor_id_option PASSED [  6%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_filter_option PASSED   [  7%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_proxy_option PASSED    [  8%]
virtwho/api/test_esx.py::TestVirtWhoConfigforEsx::test_positive_configure_organization_list PASSED [ 10%]
virtwho/api/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_deploy_configure_by_id PASSED [ 11%]
virtwho/api/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_deploy_configure_by_script PASSED [ 12%]
virtwho/api/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_hypervisor_id_option PASSED [ 14%]
virtwho/api/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_deploy_configure_by_id SKIPPED [ 15%]
virtwho/api/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_deploy_configure_by_script SKIPPED [ 16%]
virtwho/api/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_hypervisor_id_option SKIPPED [ 17%]
virtwho/api/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_deploy_configure_by_id PASSED [ 19%]
virtwho/api/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_deploy_configure_by_script PASSED [ 20%]
virtwho/api/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_hypervisor_id_option PASSED [ 21%]
virtwho/api/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_deploy_configure_by_id PASSED [ 23%]
virtwho/api/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_deploy_configure_by_script PASSED [ 24%]
virtwho/api/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_hypervisor_id_option PASSED [ 25%]
virtwho/api/test_xen.py::TestVirtWhoConfigforXen::test_positive_deploy_configure_by_id PASSED [ 26%]
virtwho/api/test_xen.py::TestVirtWhoConfigforXen::test_positive_deploy_configure_by_script PASSED [ 28%]
virtwho/api/test_xen.py::TestVirtWhoConfigforXen::test_positive_hypervisor_id_option PASSED [ 29%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_deploy_configure_by_id PASSED [ 30%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_deploy_configure_by_script PASSED [ 32%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_debug_option PASSED    [ 33%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_interval_option PASSED [ 34%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_hypervisor_id_option PASSED [ 35%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_filter_option PASSED   [ 37%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_proxy_option PASSED    [ 38%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_rhsm_option PASSED     [ 39%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_post_hypervisors PASSED [ 41%]
virtwho/cli/test_esx.py::TestVirtWhoConfigforEsx::test_positive_foreman_packages_protection PASSED [ 42%]
virtwho/cli/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_deploy_configure_by_id PASSED [ 43%]
virtwho/cli/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_deploy_configure_by_script PASSED [ 44%]
virtwho/cli/test_hyperv.py::TestVirtWhoConfigforHyperv::test_positive_hypervisor_id_option PASSED [ 46%]
virtwho/cli/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_deploy_configure_by_id SKIPPED [ 47%]
virtwho/cli/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_deploy_configure_by_script SKIPPED [ 48%]
virtwho/cli/test_kubevirt.py::TestVirtWhoConfigforKubevirt::test_positive_hypervisor_id_option SKIPPED [ 50%]
virtwho/cli/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_deploy_configure_by_id PASSED [ 51%]
virtwho/cli/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_deploy_configure_by_script PASSED [ 52%]
virtwho/cli/test_libvirt.py::TestVirtWhoConfigforLibvirt::test_positive_hypervisor_id_option PASSED [ 53%]
virtwho/cli/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_deploy_configure_by_id PASSED [ 55%]
virtwho/cli/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_deploy_configure_by_script PASSED [ 56%]
virtwho/cli/test_rhevm.py::TestVirtWhoConfigforRhevm::test_positive_hypervisor_id_option PASSED [ 57%]
virtwho/cli/test_xen.py::TestVirtWhoConfigforXen::test_positive_deploy_configure_by_id PASSED [ 58%]
virtwho/cli/test_xen.py::TestVirtWhoConfigforXen::test_positive_deploy_configure_by_script PASSED [ 60%]
virtwho/cli/test_xen.py::TestVirtWhoConfigforXen::test_positive_hypervisor_id_option PASSED [ 61%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_deploy_configure_by_id PASSED [ 62%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_deploy_configure_by_script PASSED [ 64%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_debug_option PASSED     [ 65%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_interval_option PASSED  [ 66%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_hypervisor_id_option PASSED [ 67%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_filtering_option PASSED [ 69%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_proxy_option PASSED     [ 70%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_virtwho_roles PASSED    [ 71%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_virtwho_configs_widget PASSED [ 73%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_delete_configure PASSED [ 74%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_virtwho_reporter_role PASSED [ 75%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_virtwho_viewer_role PASSED [ 76%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_virtwho_manager_role PASSED [ 78%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_overview_label_name PASSED [ 79%]
virtwho/ui/test_esx.py::TestVirtwhoConfigforEsx::test_positive_last_checkin_status PASSED [ 80%]
virtwho/ui/test_hyperv.py::TestVirtwhoConfigforHyperv::test_positive_deploy_configure_by_id PASSED [ 82%]
virtwho/ui/test_hyperv.py::TestVirtwhoConfigforHyperv::test_positive_deploy_configure_by_script PASSED [ 83%]
virtwho/ui/test_hyperv.py::TestVirtwhoConfigforHyperv::test_positive_hypervisor_id_option PASSED [ 84%]
virtwho/ui/test_kubevirt.py::TestVirtwhoConfigforKubevirt::test_positive_deploy_configure_by_id PASSED [ 85%]
virtwho/ui/test_kubevirt.py::TestVirtwhoConfigforKubevirt::test_positive_deploy_configure_by_script PASSED [ 87%]
virtwho/ui/test_kubevirt.py::TestVirtwhoConfigforKubevirt::test_positive_hypervisor_id_option PASSED [ 88%]
virtwho/ui/test_libvirt.py::TestVirtwhoConfigforLibvirt::test_positive_deploy_configure_by_id PASSED [ 89%]
virtwho/ui/test_libvirt.py::TestVirtwhoConfigforLibvirt::test_positive_deploy_configure_by_script PASSED [ 91%]
virtwho/ui/test_libvirt.py::TestVirtwhoConfigforLibvirt::test_positive_hypervisor_id_option PASSED [ 92%]
virtwho/ui/test_rhevm.py::TestVirtwhoConfigforRhevm::test_positive_deploy_configure_by_id PASSED [ 93%]                                                                                      
virtwho/ui/test_rhevm.py::TestVirtwhoConfigforRhevm::test_positive_deploy_configure_by_script PASSED [ 94%]                                                                                   
virtwho/ui/test_rhevm.py::TestVirtwhoConfigforRhevm::test_positive_hypervisor_id_option PASSED [ 96%]                                                                                        
virtwho/ui/test_xen.py::TestVirtwhoConfigforXen::test_positive_deploy_configure_by_id PASSED [ 97%]                                                                                         
virtwho/ui/test_xen.py::TestVirtwhoConfigforXen::test_positive_deploy_configure_by_script PASSED [ 98%]                                                                                       
virtwho/ui/test_xen.py::TestVirtwhoConfigforXen::test_positive_hypervisor_id_option PASSED [100%]                                                                                           

---------------------------------------------- generated xml file: /home/hkx303/Documents/CI/robottelo/robottelo/tests/foreman/test_report.xml -----------------------------------------------
=================================================================== 72 passed, 6 skipped, 95 warnings in 23748.32 seconds ====================================================================
```

### CODE REVIEW

File: robottelo/config/virtwho.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+import logging.config
+import os
+from configparser import ConfigParser
+from configparser import NoOptionError
+from configparser import NoSectionError
+
+from robottelo.config import casts
+
+LOGGER = logging.getLogger(__name__)
+SETTINGS_FILE_NAME = 'hypervisor.properties'
```
Comment (eko999) [inline robottelo/config/virtwho.py:11]: we may rename the file to 'virtwho.properties'

File: robottelo/config/virtwho.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
+        self.guest = None
+        self.guest_port = None
+        self.guest_username = None
+        self.guest_password = None
+        # SKU Information
+        self.sku_vdc_physical = None
+        self.sku_vdc_virtual = None
```
Comment (eko999) [inline robottelo/config/virtwho.py:132]: we may create a new section in virthwo.properties for SKUs:
[sku]
vdc_physical=
vdc_virtual=

File: robottelo/config/virtwho.py [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+class LibvirtSettings(FeatureSettings):
+    """VirtWho settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(LibvirtSettings, self).__init__(*args, **kwargs)
+        # Hypervisor Information
+        self.hypervisor_type = None
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
```
Comment (eko999) [inline robottelo/config/virtwho.py:349]: password option is not available for libvirt mode

File: robottelo/config/virtwho.py [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+class KubevirtSettings(FeatureSettings):
+    """VirtWho settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(KubevirtSettings, self).__init__(*args, **kwargs)
+        # Hypervisor Information
+        self.hypervisor_type = None
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
```
Comment (eko999) [inline robottelo/config/virtwho.py:405]: username and password are not available for kubevirt mode

File: robottelo/config/virtwho.py [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+        if not all(mandatory):
+            validation_errors.append(
+                '[kubevirt] hypervisor_type, hypervisor_server, hypervisor_username, '
+                'hypervisor_password, hypervisor_config_file, guest , guest_username, '
+                'guest_password, sku_vdc_physical, sku_vdc_virtual options must be provided.'
+            )
+        return validation_errors
+
+
+class HypervisorSettings(object):
```
Comment (eko999) [inline robottelo/config/virtwho.py:455]: we may rename the class name to 'VirtwhoSettings'

File: tests/foreman/virtwho/cli/test_esx.py [hunk 6]
```diff
@@ -34,9 +35,12 @@
 from robottelo.virtwho_utils import get_configure_file
 from robottelo.virtwho_utils import get_configure_option
 from robottelo.virtwho_utils import hypervisor_json_create
+from robottelo.virtwho_utils import hypervisorsettings
 from robottelo.virtwho_utils import virtwho_package_locked
 from robottelo.virtwho_utils import VIRTWHO_SYSCONFIG
 
+HYPERVISOR_TYPE = 'esx'
```
Comment (eko999) [inline tests/foreman/virtwho/cli/test_esx.py:42]: hypervisor_type is defined by hypervisorsettings.esx.hypervisor_type in form_data

File: robottelo/config/virtwho.py (hunk 1 above)
Comment (hkx303) [reply robottelo/config/virtwho.py]: Ok, changed.

File: robottelo/config/virtwho.py (hunk 2 above)
Comment (hkx303) [reply robottelo/config/virtwho.py]: Yes, that is a good idea.

File: robottelo/config/virtwho.py (hunk 3 above)
Comment (hkx303) [reply robottelo/config/virtwho.py]: I deleted it

File: robottelo/config/virtwho.py (hunk 4 above)
Comment (hkx303) [reply robottelo/config/virtwho.py]: Also deleted them

File: robottelo/config/virtwho.py (hunk 5 above)
Comment (hkx303) [reply robottelo/config/virtwho.py]: OK

File: tests/foreman/virtwho/cli/test_esx.py (hunk 6 above)
Comment (hkx303) [reply tests/foreman/virtwho/cli/test_esx.py]: Use the form_date['hypervisor_type'] instead of it.

File: robottelo/virtwho_utils.py [hunk 7]
```diff
@@ -8,9 +8,12 @@
 from robottelo.cli.host import Host
 from robottelo.cli.virt_who_config import VirtWhoConfig
 from robottelo.config import settings
+from robottelo.config.virtwho import VirtwhoSettings
 from robottelo.constants import DEFAULT_ORG
 
 VIRTWHO_SYSCONFIG = "/etc/sysconfig/virt-who"
+virtwhosettings = VirtwhoSettings()
```
Comment (eko999) [inline robottelo/virtwho_utils.py:15]: Could we change this to a short name, such as:
`virtwho = VirtwhoSettings()`

Comment (hkx303) [reply robottelo/virtwho_utils.py]: Changed. And also tested all cases, updated the test result in the description

**Review by eko999 — APPROVED** [review]
Thanks for the big improvement :+1:

**Pull request merged**